src/contractor/contractor.cpp

```diff
@@ -95,10 +98,11 @@ EdgeWeight getNewWeight(IterType new_speed_iter, const double segment_length, Ed
     {
         auto newSecs = new_segment_weight / 10;
         auto oldSecs = oldWeight / 10;
+        auto speed_file = segment_speed_filenames.at(new_speed_iter->speed_source.source - 1);
```

Comment:
I think not. `0`th source is reserved for lua profile set defaults, and where we assign this `source` value [only happens during edge weight updates](https://github.com/Project-OSRM/osrm-backend/blob/861314256ebaa0ce4ec5c1e5259da401d7edf584/src/contractor/contractor.cpp#L233) where `source` is advanced 1 to account for lua.